independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Types.java

```diff
@@ -212,6 +237,14 @@ static Type getProviderType(ClassInfo classInfo) {
             Type[] typeParams = new Type[typeParameters.size()];
             for (int i = 0; i < typeParameters.size(); i++) {
                 typeParams[i] = resolvedTypeParameters.get(typeParameters.get(i));
+                // this should only be the case for producers; wildcard is not a legal bean type
+                // see https://docs.jboss.org/cdi/spec/2.0/cdi-spec.html#legal_bean_types
+                if (typeParams[i].kind().equals(Kind.WILDCARD_TYPE) && (producerField != null || producerMethod != null)) {
+                    LOGGER.info("Producer " + (producerMethod == null ? "field " : "method ") +
```

Comment:
I wonder if we should skip the type as well, ie. do not add it to the set of types. This is what we do in Weld, right?